sdk/samples/iot/hub/src/paho_iot_hub_twin_example.c

```diff
@@ -558,15 +558,14 @@ static az_result update_property(az_span desired_payload)
       // Move to the value token
       AZ_RETURN_IF_FAILED(az_json_parser_next_token(&json_parser));
 
-      // TODO: Change back to int32_t once that is supported.
-      uint32_t temp_property_value = 0;
-      AZ_RETURN_IF_FAILED(az_json_token_get_uint32(&json_parser.token, &temp_property_value));
+      int32_t temp_property_value = 0;
+      AZ_RETURN_IF_FAILED(az_json_token_get_int32(&json_parser.token, &temp_property_value));
```

Comment:
change to
AZ_RETURN_IF_FAILED(az_json_token_get_int32(&json_parser.token, &reported_property_value));

?

Reply:
What was the intention of having the `temp_property_value` in the first place?

Reply:
Casting to int32_t.  Had to initially work with an uint32_t.  Prior to that I think we were using double and then casting to int.